src/_guides/libraries/library-tour.md

```diff
@@ -1035,6 +1035,38 @@ For more information, see
 [Exceptions](/guides/language/language-tour#exceptions)
 (in the language tour) and the [Exception API reference.][Exception]
 
+### Weak references and finalizers
+Dart is a [garbage-collected][] language,
+which means that any Dart object
+that isn't referenced by at least one other object
+will eventually be disposed by the garbage collector. 
+This default behavior may not be desireable in
+some scenarios involving native resources or 
+if the target object cannot be sub-classed.
+
+A [WeakReference][]
+stores a reference to the target object
+without increasing the reference count
+considered by the garbage collector.
+Another option is to use an [Expando][]
+to add properties to an object
+without affecting how they are garbage-collected.
+
+A [Finalizer][] can be used to execute a callback function
+after an object is no longer referenced.
+It is not guaranteed to execute this callback, however.
+
+A [NativeFinalizer][]
+provides stronger guarantees
+for interacting with native code using [dart:ffi][].
+Its callback will be called at least once
+after the object is no longer referenced.
+It can be used to close native resources
```

Comment:
```suggestion
for interacting with native code using [dart:ffi][];
its callback is invoked at least once
after the object is no longer referenced.
Also, it can be used to close native resources
```